services/products.js

```diff
@@ -620,17 +628,26 @@ const setProduct = async (req) => {
     if (!product) throw NSErrors.ProductNotFound;
     // On met à jour le slug du produit
     if (req.body.autoSlug) req.body._slug = `${utils.slugify(req.body.name)}-${req.body.id}`;
-    return new Promise((resolve, reject) => {
-        product.updateData(req.body, async (err, result) => {
-            if (err && err.message === 'slug trop court') {
-                reject(NSErrors.ProductUpdateSlugError);
-            } else if (err) {
-                reject(NSErrors.ProductUpdateError);
+    await checkSlugExist(req.body);
+    const result = await product.updateData(req.body);
+    await ProductsPreview.deleteOne({code: req.body.code});
+    await Products.findOne({code: result.code}).populated(['bundle_sections.products._id']);
+};
+
+const checkSlugExist = async (doc) => {
+    const arg = {$or: []};
+    if (doc.translation) {
+        for (const lang of Object.entries(doc.translation)) {
```

Comment:
instead of :
```js
for (const lang of Object.entries(doc.translation)) {
  if (doc.translation[lang[0]]) {
    arg.$or.push({[`translation.${lang[0]}.slug`]: doc.translation[lang[0]].slug});
  }
}
```
you could write it like that :
```js
// descructuring resulting array from Object.entries
for (const [lang] of Object.entries(doc.translation)) {
  if (doc.translation[lang]) {
    arg.$or.push({[`translation.${lang}.slug`]: doc.translation[lang].slug});
  }
}
```